### PR TITLE
Add support for user to start the email verification

### PR DESCRIPTION
**Pull request by norkans7**
Verify Email button on the account page in the edit user formax summary
Monosnap Account 2023-11-02 14-08-19

uses posterize to send them email and get the success message

Monosnap Email verification 2023-11-02 14-09-00

### CODE REVIEW

**Comment by norkans7** [comment]
Monosnap Account 2023-11-07 12-47-22

**Comment by rowanseymour** [comment]
1. I'm not sure we need a verified button beside verified email addresses - because we want to get to a point where verified is the norm
2. The user shouldn't need to click to the "Send Verification Email" view and then click again to send the actual email.. just have the first button send the email

**Comment by norkans7** [comment]
We have the button for unverified emails and we show verified label for those verified emails

Monosnap Account 2023-11-10 14-06-00

Should I remove the verified label below?
Monosnap Monosnap Account 2023-11-10 14-05-25 png 2023-11-10 14-08-21

2) I switched to use posterize and a post only view

File: temba/orgs/views.py [hunk 1]
```diff
@@ -877,6 +879,32 @@ def post_save(self, obj):
...
+        class Form(forms.ModelForm):
+            class Meta:
+                model = User
+                fields = ()
+
+        form_class = Form
+        submit_button_name = _("Send Verification Email")
+        menu_path = "/settings/account"
+        success_url = "@orgs.user_account"
+        success_message = _("Verification email sent")
```
Comment (norkans7) [inline temba/orgs/views.py:892]: @rowanseymour you mentioned we should stop using the success message but in this case we use posterize and I am not sure what will be a better way for feedback to the user.

Comment (rowanseymour) [reply temba/orgs/views.py:892]: yeah in this case it's ok

File: templates/orgs/user_edit.html [hunk 2]
```diff
@@ -19,10 +19,25 @@
...
+  <div class="flex items-center">
+    <div class="flex-grow mr-4 api_token_container">
+      {% blocktrans with email=user.username %}
+      Your email address is <b>{{email}}</b>
+      {% endblocktrans %}
+      {% if user.settings.email_status == 'V' %}
+        <temba-label icon="icon.org_verified" class="mr-2">
+          {% trans "Verified" %}
+        </temba-label>
+      {% endif %}
```
Comment (norkans7) [inline templates/orgs/user_edit.html:31]: @rowanseymour 
should I remove this label?

Monosnap Monosnap Account 2023-11-10 14-05-25 png 2023-11-10 14-08-21

Comment (rowanseymour) [reply templates/orgs/user_edit.html:31]: I think so... the goal here is to make verified the norm.. make it part of the signup process

Comment (norkans7) [reply templates/orgs/user_edit.html]: Done

**Comment by ericnewcomer** [comment]
Why is the view to trigger the email a page and not a modal?

**Comment by norkans7** [comment]
Currently we just have a button that posterizes to send the email, so not a page
Monosnap Account 2023-11-10 14-06-00

Click on the button and the email is sent.

I can remove the old screenshots if they now confusing

**Comment by ericnewcomer** [comment]
Ah, sorry for the confusion! Makes sense.

File: templates/orgs/user_send_verification_email.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
+{% extends "smartmin/update.html" %}
```
Comment (rowanseymour) [inline templates/orgs/user_send_verification_email.html:1]: is this template used?

File: temba/orgs/models.py [hunk 4]
```diff
@@ -213,6 +213,40 @@ def set_email_status(self, status: str):
...
+        verification_secret = self.settings.email_verification_secret
+        if not verification_secret:
+            verification_secret = generate_secret(64)
+
+            self.settings.email_verification_secret = verification_secret
+            self.settings.save(update_fields=("email_verification_secret",))
+
+        # no=op if we do not know the email
+        if not self.email:  # pragma: needs cover
+            return
```
Comment (rowanseymour) [inline temba/orgs/models.py:237]: this only applies to the superuser if it exists so don't need to check this

File: temba/orgs/models.py [hunk 5]
```diff
@@ -213,6 +213,40 @@ def set_email_status(self, status: str):
...
+
+        org = self.get_orgs().first()
+
+        subject = _("%(name)s Email Verification") % org.branding
+        template = "orgs/email/email_verification"
+        to_email = self.email
```
Comment (rowanseymour) [inline temba/orgs/models.py:243]: doesn't need to be a variable

File: temba/orgs/tasks.py [hunk 6]
```diff
@@ -28,6 +28,12 @@ def send_invitation_email_task(invitation_id):
     invitation.send_email()
 
 
+@shared_task
+def send_user_verification_email_task(user_id):
+    user = User.objects.get(pk=user_id)
```
Comment (rowanseymour) [inline temba/orgs/tasks.py:33]: id=

File: temba/orgs/models.py [hunk 7]
```diff
@@ -213,6 +213,35 @@ def set_email_status(self, status: str):
...
+
+        org = self.get_orgs().first()
+
+        subject = _("%(name)s Email Verification") % org.branding
+        template = "orgs/email/email_verification"
+
+        context = dict(org=org, now=timezone.now(), branding=org.branding, secret=verification_secret)
+        context["subject"] = subject
+
+        send_template_email(self.email, subject, template, context, org.branding)
```
Comment (rowanseymour) [inline temba/orgs/models.py:243]: Sometimes it makes sense to have this pattern of model method invoking task, task calling model method, but here it just feels like we're adding unhelpful complexity. We have...

```
User.send_verification
User.send_verification_email
send_user_verification_email_task
```

Let's just have a task called `send_user_verification_email` which is called by the view.

File: templates/orgs/user_edit.html [hunk 8]
```diff
@@ -19,10 +19,20 @@
   </div>
 {% endblock fields %}
 {% block summary %}
-  {% blocktrans with email=user.username %}
-    Your email address is <b>{{email}}</b>.
-
-  {% endblocktrans %}
+  <div class="flex items-center">
+    <div class="flex-grow mr-4 api_token_container">
+      {% blocktrans with email=user.username %}
```
Comment (rowanseymour) [inline templates/orgs/user_edit.html:24]: trimmed

**Review by rowanseymour — CHANGES_REQUESTED** [review]

**Pull request merged**